### PR TITLE
More conforming to C++ standards

### PR DESCRIPTION
**Pull request by gongminmin**
1. Can't mix char* and const char*
2. Remove redundant const and std::move
3. Use standard types and macros to replace platform dependent ones

### CODE REVIEW

**Review by llvm-beanz — APPROVED** [review]
These all look like reasonable changes to me.

Is there a set of compiler warnings you enabled when developing this patch?
Might be worth enabling extra warnings in the future to prevent regression.

**Review by python3kgae — APPROVED** [review]

**Review by pow2clk — COMMENTED** [review]
Looks good. 

I don't see any example of some of the things mentioned in the commit message. 

2. Use this-> to represent member from parent template classes
I don't see anything similar to this.

3. Can't do double implicit conversion

Maybe not this either? There are some places that lose casting. I don't know if this is in reference to double type casting or two casts that form a (double)

Perhaps these comments represent an earlier version of the change?

**Comment by gongminmin** [comment]
> Looks good.
> 
> I don't see any example of some of the things mentioned in the commit message.
> 
>     2. Use this-> to represent member from parent template classes
>        I don't see anything similar to this.
> 
>     3. Can't do double implicit conversion
> 
> 
> Maybe not this either? There are some places that lose casting. I don't know if this is in reference to double type casting or two casts that form a (double)
> 
> Perhaps these comments represent an earlier version of the change?

Right. 2) and 3) are already fixed in main branch in the last 15 months. I've updated the commit message and the PR description.

**Review by pow2clk — APPROVED** [review]
Thanks for updating the commit message!

File: tools/dxexp/dxexp.cpp [hunk 1]
```diff
@@ -296,7 +296,7 @@ static HRESULT PrintAdapters() {
     hr = e.m_hr;
     json_printf("%c { \"err\": \"unable to print information for adapters - "
                 "0x%08x\" }\n",
-                comma, hr);
+                comma, (unsinged int)hr);
```
Comment (python3kgae) [inline tools/dxexp/dxexp.cpp:299]: need to be unsigned

But doesn't have access to push fix:(

Comment (gongminmin) [reply tools/dxexp/dxexp.cpp]: Fixed.

**Pull request merged**